test/common/src/main/java/io/camunda/zeebe/spring/test/AbstractZeebeTestExecutionListener.java

```diff
@@ -22,33 +25,28 @@ public class AbstractZeebeTestExecutionListener {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
 
-  private ZeebeClient zeebeClient;
+  public <T extends ZeebeTestEngine> void setupWithZeebeEngine(TestContext testContext, T zeebeEngine, final UnaryOperator<T> startCallback) {
 
-  /**
-   * Registers the ZeebeEngine for test case in relevant places and creates the ZeebeClient
-   */
-  public void setupWithZeebeEngine(TestContext testContext, ZeebeTestEngine zeebeEngine) {
+    if (Optional.ofNullable(testContext.getAttribute("NOT_STARTED")).map(o -> (Boolean) o).orElse(false)) {
```

Comment:
Small thing, but confuses me: I would name the attribute "STARTED" to avoid a double-boolean logic :-)

That's said I wonder why when called the first time, this is initialized with "NOT_STARTED" = False = STARTED? That means the engine is not set on the proxy the first time. I double checked and I think that is because you now initialize the proxy immediately when creating. This was cautiously changed - right? 

IF I understand it correctly, the first engine is now created on bean creation time - and then in the first test method nothing happens. In the following test methods, the engine is swapped. 

Engine creation is then not done in the before/after methods directly, but by the callback handed in - so you can control here when it is created. Is that a correct description of your idea?

Reply:
@aivinog1

Reply:
Hey @berndruecker! 
> That means the engine is not set on the proxy the first time. I double checked and I think that is because you now initialize the proxy immediately when creating. This was cautiously changed - right?
> IF I understand it correctly, the first engine is now created on bean creation time - and then in the first test method nothing happens. In the following test methods, the engine is swapped.
> Engine creation is then not done in the before/after methods directly, but by the callback handed in - so you can control here when it is created. Is that a correct description of your idea?
Yes, that is true, I make it intentionally and this is true about my idea :)